legend-engine-xt-documentStore-executionPlan-connection/src/main/java/org/finos/legend/engine/plan/execution/stores/nonrelational/client/NonRelationalClient.java

```diff
@@ -24,4 +24,6 @@
     // NonRelationalClient buildNonRelationalClient(Identity identity, String connectionUri);
 
     List<String> executeNativeQuery(String mongoQuery);
+
+    void shutDown();
```

Comment:
should start Server also be here?? because it seems a bit inconsistent.  Worth checking what Connection interface providdes

Reply:
With Connection ,a Statement is created and that is what is used to execute SQL queries. (RelationalExecutor.java line 323 for example) On the relational side, HikariCP is used to create HikariDataSources and Hikari manages the connection pools. There is a ConnectionStateManagerPOJO concept but if we want to follow that it seems this is tied to the usage of HikariDataSource and DataSourceSpecification. 

To me, shutDown is needed for internal clean up prior to no longer needing the NonRelationalClient, if we initialise the client I don't see why we need an explicit initialise/begin/start() method, but we can add it. We will need to think about Connection pooling at some point but for our first cut here I wouldn't add too much complexity/management.

Reply:
the Mongo drivers do that, so we don't need any other library like hikari for that purpose.